lib/IDE/Refactoring.cpp

```diff
@@ -3887,7 +3889,1442 @@ bool RefactoringActionConvertToComputedProperty::performChange() {
   EditConsumer.accept(SM, ReplaceCharSourceRange, DeclBuffer.str());
   return false; // success
 }
-}// end of anonymous namespace
+
+namespace asyncrefactorings {
+
+// TODO: Should probably split the refactorings into separate files
+
+/// Whether the given parameter name identifies a completion callback
+bool isCompletionHandlerName(StringRef Name) {
+  return Name.startswith("completion") || Name.contains("Completion") ||
+         Name.contains("Complete");
+}
+
+/// Whether the given type is the stdlib Result type
+bool isResultType(Type Ty) {
+  if (!Ty)
+    return false;
+  if (auto *NTD = Ty->getAnyNominal())
+    return NTD == NTD->getASTContext().getResultDecl();
+  return false;
+}
+
+/// Whether the given type is (or conforms to) the stdlib Error type
+bool isErrorType(Type Ty) {
+  if (!Ty)
+    return false;
+
+  auto *ED = Ty->getASTContext().getErrorDecl();
+  if (auto *NTD = Ty->getAnyNominal()) {
+    if (NTD == ED)
+      return true;
+    for (auto Conformance : NTD->getAllConformances()) {
+      if (Conformance->getProtocol() == ED)
+        return true;
+    }
+  }
+
+  return false;
+}
+
+/// Whether the given type is (or conforms to) the stdlib Never type
+bool isNeverType(Type Ty) {
```

Comment:
I think you can just do Ty->isUninhabited() to check this (Never is just an empty enum).